osu.Game/Screens/Select/Carousel/CarouselGroupEagerSelect.cs

```diff
@@ -101,8 +101,7 @@ private void attemptSelection()
 
         protected virtual CarouselItem GetNextToSelect()
         {
-            return Children.Skip(lastSelectedIndex).FirstOrDefault(i => !i.Filtered.Value) ??
-                   Children.Reverse().Skip(InternalChildren.Count - lastSelectedIndex).FirstOrDefault(i => !i.Filtered.Value);
+            return InternalChildren.OrderBy(i => Math.Abs(InternalChildren.IndexOf(i) - lastSelectedIndex)).FirstOrDefault(i => !i.Filtered.Value);
```

Comment:
``OrderBy`` is probably wasteful. Could just get first visible item both directions and check which one is closer to last selection.